src/app/portlets/content-types/fields/content-type-fields-add-row/content-type-fields-add-row.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { MockMessageService } from './../../../../test/message-service.mock';
+
+describe('ContentTypeFieldsAddRowComponent', () => {
+    let comp: ContentTypeFieldsAddRowComponent;
+    let fixture: ComponentFixture<ContentTypeFieldsAddRowComponent>;
+    let de: DebugElement;
+    let el: HTMLElement;
+    let testHotKeysMock: TestHotkeysMock;
+
+    const messageServiceMock = new MockMessageService({
+        'contenttypes.content.add_rows': 'Add Rows'
+    });
+
+    beforeEach(async(() => {
+        testHotKeysMock = new TestHotkeysMock();
+
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                ContentTypeFieldsAddRowComponent
+            ],
+            imports: [
+                TooltipModule,
+                BrowserAnimationsModule
+            ],
+            providers: [
+                { provide: HotkeysService, useValue: testHotKeysMock },
+                { provide: MessageService, useValue: messageServiceMock }
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(ContentTypeFieldsAddRowComponent);
+        comp = fixture.componentInstance;
+        de = fixture.debugElement;
+        el = de.nativeElement;
+    }));
+
+    it('should display the add rows button by default', () => {
+        const addRowContainer = de.nativeElement.querySelector('.dot-add-rows__container');
+
+        fixture.detectChanges();
+
+        expect(addRowContainer.classList.contains('dot-add-rows__add')).toEqual(true);
+    });
+
+    it('should display a row after click on AddRows button', () => {
+        fixture.detectChanges();
+        const lis = de.queryAll(By.css('li'));
+
+        const addButton = de.nativeElement.querySelector('.dot-add-rows-button__container button');
+        spyOn(comp, 'selectColumnState');
+        addButton.click();
+
+        expect(lis.length).toEqual(4);
+    });
+
+    it('should escape to previous step', () => {
+        comp.selectColumnState();
+
+        testHotKeysMock.callback(['esc']);
+
+        expect(comp.rowState).toBe('add');
+    });
+
+    it('should be able to use left keyboard to highlight columns', () => {
+        comp.selectColumnState();
+
+        for (let i = 0; i < 5; i++) {
+            testHotKeysMock.callback(['left']);
+            expect(comp.selectedLi).not.toBeLessThan(0);
+        }
+    });
+
+    it('should be able to use right keyboard to highlight columns', () => {
+        comp.selectColumnState();
+
+        for (let i = 0; i < 5; i++) {
+            testHotKeysMock.callback(['right']);
+            expect(comp.selectedLi).not.toBeGreaterThan(3);
+        }
+    });
+
+    it('should select columns number after click on li', () => {
+        fixture.detectChanges();
+        const lis = de.queryAll(By.css('li'));
+
+        spyOn(comp.selectColums, 'emit');
+        comp.selectColums.subscribe(col => {
+            expect(col).toEqual(2);
+        });
+
+        lis[1].nativeElement.click(2);
```

Comment:
Why are you passing a 2 to the click? click method doesn't have params: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click

I don't think this test it's testing what you expect, I change the 2 for a 3 in the click and still pass, you need to revisit this.